mobile-save-for-later-user-deletion/conf/cfn.yaml

```diff
@@ -0,0 +1,162 @@
+          PolicyDocument:
+            Statement:
+              Effect: Allow
+              Action:
+                - lamda:InvokeFunction
+              Resource: "*"
+        - PolicyName: sqs-read
+          PolicyDocument:
+            Statement:
+              Effect: Allow
+              Action:
+                - sqs:ReceiveMessage
+                - sqs:DeleteMessage
+                - sqs:GetQueueAttributes
+              Resource:
+                - !GetAtt UserIdDeleteQueue.Arn
+
+  UserIdDeletionQueuePolicy:
+    Type: AWS::SQS::QueuePolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: "*"
+            Action:
+              - sqs:SendMessage
+            Resource: !GetAtt UserIdDeleteQueue.Arn
+            Condition:
+              ArnEquals:
+                aws:SourceArn: !Sub ${IdentityDeleteUserSnsTopicArnBase}-${Stage}
+
+      Queues:
+        - !Ref UserIdDeleteQueue
+
+
+  UserIdDeleteQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub UserIdDeleteQueue-${Stage}
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt UserIdDeletionDeadLetterQueue.Arn
+        maxReceiveCount: 3
+
+  UserIdDeletionDeadLetterQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub UserIdDeleteQueue-deadletter-${Stage}
+
+  UserDeletionAlarmTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: !Sub UserDeletionQueueAlert-${Stage}
+      Subscription:
+        - Endpoint: mobile.server.side@theguardian.com
+          Protocol: email
+
+
+  UserDeletionQueueDepthAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: "Ensure that the user ids are being deleted by the user deletion lambda"
+      Namespace: !Sub SaveForLater/${Stage}
+      MetricName: ApproxNumberOfPendingUserIdDeletions
+      Dimensions:
+        - Name: QueueName
+          Value: !GetAtt 'UserIdDeleteQueue.QueueName'
+      Period: 3600
+      Statistic: Sum
+      EvaluationPeriods: 1
+      ComparisonOperator: GreaterThanThreshold
+      Threshold: 10
+      AlarmActions:
+        - Ref: UserDeletionAlarmTopic
+      InsufficientDataActions:
+        - Ref: UserDeletionAlarmTopic
+          
+  UserDeletionLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Sub ${App}-${Stage}
+      Code:
+        S3Bucket:
+          Ref: DeployBucket
+        S3Key: !Sub ${Stack}/${Stage}/${App}/${App}.jar
+      Environment:
+        Variables:
+          App: !Sub ${App} 
+          Stack: !Sub ${Stack} 
+          Stage: !Sub ${Stage}
+          SaveForLaterApp: !Sub ${SaveForLaterApp}
+      Description: Lamba that deletes saved for later data for deleted users
+      Handler: com.gu.sfl.userdeletion.UserDeletionLambda::handler
+      MemorySize: 384
+      Role: !GetAtt UserDeletionRole.Arn
+      Runtime: java8
+      Timeout: 20
```

Comment:
20s seems low, could this be high? 300s? I'm guessing we're not expecting much traffic

Reply:
15 to 20 uses a day, according to identity